src/api/java/com/minecolonies/api/colony/requestsystem/resolver/IRequestResolverProvider.java

```diff
@@ -0,0 +1,24 @@
+package com.minecolonies.api.colony.requestsystem.resolver;
+
+import com.google.common.collect.ImmutableCollection;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+
+/**
+ * Interface used to describe a class that provides resolvers.
+ * Should be put on Buildings, Citizens etc who can resolve certain requests.
+ * <p>
+ * If a provider is added to his or her colony
+ */
+public interface IRequestResolverProvider
```

Comment:
This class is not supposed to be in this PR yet.
Just the Factories part and the IToken part (and there respective implementations)